tensorflow/python/framework/tensor_util.py

```diff
@@ -368,7 +368,9 @@ def make_tensor_proto(values, dtype=None, shape=None, verify_shape=False):
       np_dt = dtype.as_numpy_dtype
     else:
       np_dt = None
-    if np.prod(shape) == 0:
+    # if shape is None, numpy.prod returns None when dtype is not set, but raise
+    # exception when dtype is set to np.int64
+    if not(shape is None) and np.prod(shape, dtype=np.int64) == 0:
```

Comment:
I would write this as 

if shape is not None and np.prod(shape, dtype=np.int64) == 0: